compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -584,17 +563,29 @@ class ComposeScene internal constructor(
     }
 
     private fun processPress(event: PointerInputEvent) {
-        val owner = hoveredOwner(event)
-        if (focusedOwner.isAbove(owner)) {
-            focusedOwner?.onDismissRequest?.invoke()
-        } else {
-            owner?.processPointerInput(event)
-            pressOwner = owner
+        val owner = reversedOwners {
+            for (owner in it) {
+                if (owner.isHovered(event)) {
+                    // Stop once the position of in bounds of the owner
+                    return@reversedOwners owner
+                }
+                owner.onClickOutside?.invoke()
+                if (owner == focusedOwner) {
+                    // Stop if it's in focus, do not pass the event to hovered owner
+                    return@processPress
```

Comment:
can we replace this by `return@reversedOwners null` to simplify possible branching and to nullify the previous `pressOwner` (in case of multitouch)?

Reply:
I don't want to introduce changing `pressOwner` on click outside, because it's most likely break some untested cases. As discussed with @m-sasha, we most like need to revisit it later because currently we have no guarantee that for release event we sent press event to this (same) object/owner.

For now I do not see strong reason to introduce more changes in this PR

Reply:
It is an early return of `fun processPress` inside lambda.
Line below `pressOwber = owner` in this case will not be called. Is it intended?

Reply:
Yes

Reply:
Sounds reasonable to not change the bahavior.

In this case may be we can simplify the code this way?

```
    private fun processPress(event: PointerInputEvent) {
        forEachReversedOwner { owner ->
            if (owner.isHovered(event)) {
                owner.processPointerInput(event)
                pressOwner = owner
                // Stop once the position of in bounds of the owner
                return
            }
            owner.onClickOutside?.invoke()
            if (owner == focusedOwner) {
                // Stop if it's in focus, do not pass the event to hovered owner
                return
            }
        }
    }
```
We don't have 2 types of breaks here (returning a value, and exiting the main function)

Reply:
I agree with @igordmn that forEachOwnerReversed looks more consistent here.

Reply:
@igordmn No, it's not simplifying, it's again changing nulling behaviour - it won't assign null if there is no hovered owner.